How to build for android?

SO : Arch Linux
java -version
openjdk version "17.0.12" 2024-07-16
OpenJDK Runtime Environment (build 17.0.12+7)
OpenJDK 64-Bit Server VM (build 17.0.12+7, mixed mode, sharing)

How can I build for Android with these new changes?

git clone https://github.com/koreader/koreader.git
cd koreader && ./kodev fetch-thirdparty


Tried running the Android debug apk build command:
./kodev run android-arm64
Causes the error:
make ANDROID_ARCH=arm64 TARGET=android KODEBUG=1 VERBOSE= run
bash: line 1: aarch64-linux-android21-clang: command not found
bash: line 1: aarch64-linux-android21-clang: command not found
base/Makefile.defs:441: *** failed to retrieve target machine.  Stop.

Tried instead the following command with success:
./kodev build android-arm64

```      | [koreader 100%] Linking C shared library libwrap-mupdf.so
▸100% | Installing 'koreader'
[*] Install update once marker
[*] Install C++ runtime...
[*] Install plugins
[*] Install resources
[*] Install data files
[*] Clean up, remove unused files for releases
make: 
unlink: base/thirdparty/kpvcrlib/crengine/cr3gui/data/tessdata: Is a directory
make: 
unlink: base/thirdparty/kpvcrlib/crengine/cr3gui/data/dict: Is a directory
```

Tried again 

./kodev run android-arm64

It goes pretty well until near the end, but only the NDK is downloaded and it asks to accept the licenses. However, no SDK tools are downloaded to accept these licenses—only the NDK. How can I accept the licenses?

```
Files read from disk: 953
Archive size: 19898993 bytes (19 MiB)
Everything is Ok
Starting a Gradle Daemon (subsequent builds will be faster)
Configuration on demand is an incubating feature.
Calculating task graph as no cached configuration is available for tasks: app:assemblearm64RocksDebug

> Configure project :app
WARNING: The option setting 'android.defaults.buildfeatures.buildconfig=true' is deprecated.
The current default is 'false'.
It will be removed in version 9.0 of the Android Gradle plugin.
You can resolve this warning in Android Studio via `Refactor` > `Migrate BuildConfig to Gradle Build Files`

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/home/hugleo/Desktop/koreader/koreader/platform/android/luajit-launcher/local.properties'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 36s
Configuration cache entry stored.
make: *** [make/android.mk:87: update] Error 1
```

/home/hugleo/Desktop/koreader/koreader/base/toolchain
$ ls
android-ndk-r23c  Makefile

--- a/kodev
+++ b/kodev
@@ -111,7 +111,7 @@ check_submodules() {
     # due to `git submodule status` returning an error
     # on `grep -q …` early exit (broken pipe).
     if grep -qE '^-' <<<"$(git submodule status)"; then
-        run_make fetchthirdparty
+        TARGET='' run_make fetchthirdparty
     fi
 }
 
@@ -192,6 +192,15 @@ function setup_target() {
     if [[ -z ${KODEBUG+x} && -z "${TARGET}" && "${CMD}" != 'release' ]]; then
         # For the emulator, build a debug build by default.
         KODEBUG=1
+    fi
+    # Fetch third party as needed.
+    check_submodules
+    # Automatically install Android NDK / SDK when not configured.
+    if [[ "${TARGET}" == 'android' ]]; then
+        local wanted=()
+        [[ -n "${ANDROID_NDK_HOME}${ANDROID_NDK_ROOT}" ]] || wanted+=('android-ndk')
+        [[ -n "${ANDROID_HOME}${ANDROID_SDK_ROOT}" ]] || wanted+=('android-sdk')
+        [[ ${#wanted} -eq 0 ]] || TARGET='' run_make "${wanted[@]}"
     fi
 }
 
@@ -334,10 +343,6 @@ ${TARGETS_HELP_MSG}
 "
     parse_options "${BUILD_GETOPT_SHORT}" "${BUILD_GETOPT_LONG}" '?' "$@"
     setup_target "${ARGS[0]}"
-    check_submodules
-    if [[ "${TARGET}" == 'android' && -z "${ANDROID_NDK_HOME}${ANDROID_NDK_ROOT}" ]]; then
-        TARGET='' run_make android-ndk
-    fi
     run_make ${NO_BUILD:+setup}
 }
 
@@ -376,16 +381,12 @@ ${RELEASE_TARGETS_HELP_MSG}
         esac
     done
     setup_target "${ARGS[0]}"
-    check_submodules
     if [[ "${ignore_translation}" -eq 0 ]]; then
-        if ! run_make po; then
+        if ! TARGET='' run_make po; then
             err "ERROR: failed to fetch translation."
             echo "Tip: Use --ignore-translation OPTION if you want to build a release without translation." 2>&1
             exit 1
         fi
-    fi
-    if [[ "${TARGET}" == 'android' ]] && [[ -z "${ANDROID_NDK_HOME}${ANDROID_NDK_ROOT}" || -z "${ANDROID_HOME}${ANDROID_SDK_ROOT}" ]]; then
-        TARGET='' run_make android-ndk android-sdk
     fi
     run_make ${NO_BUILD:+--assume-old=all} update
 }
@@ -575,7 +576,7 @@ ANDROID TARGET:
     fi
     # Build the final make command.
     local margs=()
-    if [[ "${target}" = android-* ]]; then
+    if [[ "${TARGET}" == 'android' ]]; then
         if [[ "$1" = *.apk ]]; then
             margs+=(ANDROID_APK="$1")
             NO_BUILD=1
@@ -640,7 +641,6 @@ $(build_options_help_msg 'BUILD' 'use existing build' '' 'default')
         esac
     done
     setup_target 'emulator'
-    check_submodules
     run_make ${NO_BUILD:+--assume-old=all} ${show_previous:+--assume-old=coverage-run} coverage${show_full:+-full}
 }
 
@@ -668,7 +668,6 @@ $(build_options_help_msg 'BUILD' 'use existing build' '' 'default')
     fi
     # The rest (custom options included) is forwarded to busted.
     setup_target 'emulator'
-    check_submodules
     run_make ${NO_BUILD:+--assume-old=all} "test${suite}" BUSTED_OVERRIDES="$(print_quoted "${OPTS[@]}" "${ARGS[@]}")"
 }
 

